R/preprocess.R

```diff
@@ -296,30 +360,37 @@ enw_preprocess_data <- function(obs, by = c(), max_delay = 20,
   )
   diff_obs[, group := new_group][, new_group := NULL]
   obs[, old_group := NULL]
+  
+  # separate obs with and without missing reference date
+  reporting_available <- diff_obs[!is.na(reference_date)]
+  reporting_missing <- diff_obs[is.na(reference_date)]
 
-  # calculate reporting matrix
-  reporting_triangle <- enw_reporting_triangle(diff_obs)
+  # calculate reporting matrix on obs with available reference date
+  reporting_triangle <- enw_reporting_triangle(reporting_available)
 
   # extract latest data
-  latest <- enw_latest_data(obs)
+  # Note: currently, only the obs with available reference date are used
+  # This should to be extended to missing reference dates to avoid bias
+  latest <- enw_latest_data(reporting_available)
```

Comment:
As implemented at the moment the filtering for reference dates in `enw_latest_data` will also need `na.rm = TRUE` I think.